charts/kargo/templates/api/deployment.yaml

```diff
@@ -97,20 +97,46 @@ spec:
 {{- end }}
             initialDelaySeconds: 5
 {{- end }}
-{{- if or .Values.kubeconfigSecrets.kargo (and .Values.api.oidc.enabled .Values.api.oidc.dex.enabled) .Values.api.tls.enabled }}
+{{- if or .Values.kubeconfigSecrets.kargo (and .Values.api.oidc.enabled .Values.api.oidc.dex.enabled) .Values.api.tls.enabled .Values.api.cabundle.enabled }}
           volumeMounts:
+          {{- if or .Values.kubeconfigSecrets.kargo (and .Values.api.oidc.enabled .Values.api.oidc.dex.enabled) .Values.api.tls.enabled }}
             - mountPath: /etc/kargo
               name: config
               readOnly: true
+          {{- end }}
+          {{- if .Values.api.cabundle.enabled }}
+            - mountPath: /etc/ssl/certs
+              name: certs
+          {{- end }}
 {{- end }}
 {{- with .Values.api.securityContext | default .Values.global.securityContext }}
           securityContext:
             {{- toYaml . | nindent 12 }}
 {{- end }}
           resources:
             {{- toYaml .Values.api.resources | nindent 12 }}
-{{- if or .Values.kubeconfigSecrets.kargo (and .Values.api.oidc.enabled .Values.api.oidc.dex.enabled) .Values.api.tls.enabled }}
+      {{- if .Values.api.cabundle.enabled }}
+      initContainers:
+      - name: parse-cabundle
+        image: docker.io/alpine/curl:latest
```

Comment:
I understand the reason for the init container, but introducing a new image may complicate upcoming efforts to harden our images. i.e. Distroless images are in Kargo's near future. I am hesitant to bring in a new image for which we will have to contend with CVEs on an ongoing basis.

_If_ we were to bring in a new image, I'd prefer it not be Alpine based, but something distroless.

But rather than contend with a new image at all, it may be better if we just modify the startup process to handle this. (Note: This would not entail a new, scripted entrypoint, because I believe you can expect our hardened image not to have a shell. We would have to integrate the functionality into the existing binary, which shouldn't be too hard.)

I'm interested to hear @hiddeco's opinion about this.

Reply:
I've thought about building a container specific to the use case for using this type of initContainer, but even then it still another container maintained by someone else.  This is also an initContainer from a time when I was using curl to download certs instead of using a configmap/secret, so open to switching out the initcontainer image used.

Enabling 'initContainers' to be specified and 'extraVolumes' and 'extraVolumeMounts' would almost enable a user to set something up themselves except they'd never really know where to mount the certs, everyone can guess its probably /etc/ssl/certs but one never really knows ... and wouldn't it be nice if every user mounting a ca-bundle ends up with the same working setup.  (reminded of the controller which needs to write to /etc/ssl/certs, and everyone trying to solve this on their on probably going to run into that issue of /etc/ssl/certs being RO after mounting a volume there)... in the end the minimal is to add 'update-ca-trust' somewhere and somehow provide a mount to add certs ... but I would hope to include the ability to split up a cert file containing multiple certs as that happens in pretty much every situation.

Reply:
I think that moving this into the Kargo image would not be too great either, as I do not think it is something that should be solved by us (but rather something we should allow to be facilitated, _however_ the user pleases).

Given this, I wonder if we can not just get away with allowing people to define their own set of mounts and init containers. Which could facilitate this, and whatever else people would like to do.

Reply:
I can see the point of that, but also can't help feeling that installing/amending a ca bundle may end up a frequent-enough concern that it would seem silly not to have a built-in way to accommodate it. I'm a yes on leaving the door open for users to do more things here, but I also don't think everyone approaching this particular problem should have to cook up their own solution to it.

Reply:
The thing about letting people do it on there own is no one can ever really know where the certificates should be placed, or whether or not creating a read-only scenario by mounting a secret or configmap will be a problem if the app itself needs to write to a location.